src/baseIdrissCrypto.ts

```diff
@@ -52,11 +53,12 @@ export abstract class BaseIdrissCrypto {
             connectionOptions.sendToAnyoneContractAddress : this.IDRISS_SEND_TO_ANYONE_CONTRACT_ADDRESS
         this.IDRISS_TIPPING_CONTRACT_ADDRESS = (typeof connectionOptions.tippingContractAddress !== 'undefined') ?
             connectionOptions.tippingContractAddress : this.IDRISS_TIPPING_CONTRACT_ADDRESS
-
+        this.TWITTER_BEARER_TOKEN = (typeof twitterApiKey !== 'undefined') ?
+            twitterApiKey : this.TWITTER_BEARER_TOKEN
```

Comment:
You don't need to check for undefined here since in the constructor arguments you expect string.

Actually to make it non-breaking change you should type it as optional:
```ts
twitterApiKey?: string;
```

I also suggest to use connection options since it is our configuration object for instantiating the class.

Reply:
This should be enough:
```ts
this.TWITTER_BEARER_TOKEN = twitterApiKey;
```